src/bin/pos.rs

```diff
@@ -0,0 +1,146 @@
+use egg::*;
```

Comment:
Would be good to add a link to the coq page on this and/or some description of what this domain is and what XH, X0, X1 are.